include/pythonAPI.h

```diff
@@ -28,18 +28,26 @@ int getTypenum(){
     return -1;
 }
 
+template <class T>
+void wrap_numpy_allocate(Mat<T>* new_mat, void* userdata, const DimInfo& dim_info){ //do we need a struct or is ndim enough?
+    new_mat->dims = new size_t[dim_info.ndim];
+    new_mat->strides = new size_t[dim_info.ndim];
+    new_mat->allocator->userdata = userdata;
+    Py_INCREF((PyObject*)userdata);
+}
+
+template <class T>
+void wrap_numpy_deallocate(Mat<T>* mat){
+    Py_DECREF((PyObject*)mat->allocator->userdata);
```

Comment:
For non-default allocator case(like here), move the delete lines for dims/strides to the deallocate function